packages/flutter/lib/src/material/notch_computers.dart

```diff
@@ -0,0 +1,158 @@
+import 'dart:math' as math;
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/rendering.dart';
+
+/// Computes notches in the outline of a shape.
+///
+/// Typically used to compute notches by a [BottomAppBar].
+abstract class NotchComputer {
```

Comment:
Although I appreciate the symmetry ComputeNotch and NotchComputer, calling a class FooComputer seems a little odd.  The "computer" suffix sounds like a noun.  Like the Kitchen Computer (https://en.wikipedia.org/wiki/Honeywell_316#Kitchen_Computer) :-).

Why not just call the class Notch?

Reply:
done

Reply:
Also renamed notch_computers.dart to notches.dart (and the test file accordingly)